app/models/patient.rb

```diff
@@ -100,6 +100,33 @@ def self.trim_urgent_patients
     end
   end
 
+  def self.contacted_since(datetime)
+    # 1) filters to all patients with a call logged as Reached patient >= the date and counts them
+    # 2) identifies how many of those patients have been marked pledge_sent this week
+
+    # returns hash
+    # {
+    #   since: datetime,
+    #   contacts: (count of patients reached)
+    #   first_contacts: (count of first time reached patients),
+    #   pledges_sent: (count of pledges sent)
+    # }
+
+    patient_reached = 0
+    patient_pledges_sent = 0
+
+
+    Patient.each do |patient|
```

Comment:
You might try out ruby's `select` here instead of iterating and incrementing -- probably a bit quicker to filter down rather than looking thru things!